solc/main.cpp

```diff
@@ -44,36 +44,36 @@ int main(int argc, char** argv)
 	{
 		cerr << "SMT logic error:" << endl;
 		cerr << boost::diagnostic_information(_exception);
-		return 1;
+		return 2;
 	}
 	catch (langutil::UnimplementedFeatureError const& _exception)
 	{
 		cerr << "Unimplemented feature:" << endl;
 		cerr << boost::diagnostic_information(_exception);
-		return 1;
+		return 2;
 	}
 	catch (langutil::InternalCompilerError const& _exception)
 	{
 		cerr << "Internal compiler error:" << endl;
 		cerr << boost::diagnostic_information(_exception);
-		return 1;
+		return 2;
 	}
 	catch (boost::exception const& _exception)
 	{
 		cerr << "Uncaught exception:" << endl;
 		cerr << boost::diagnostic_information(_exception) << endl;
-		return 1;
+		return 2;
 	}
 	catch (std::exception const& _exception)
 	{
 		cerr << "Uncaught exception:" << endl;
 		cerr << boost::diagnostic_information(_exception) << endl;
-		return 1;
+		return 2;
 	}
```

Comment:
You can actually remove these two. `diagnostic_information` does not provide any extra info over `current_exception_diagnostic_information` so the current `(...)` handler can deal with them all just fine.